test/e2e/storage/subpath.go

```diff
@@ -125,5 +125,33 @@ var _ = utils.SIGDescribe("Subpath", func() {
 			}, privilegedSecurityContext)
 			testsuites.TestBasicSubpath(f, "configmap-value", pod)
 		})
+
+	})
+
+	ginkgo.Context("Pod restart", func() {
+		var err error
+		var privilegedSecurityContext bool = false
+
+		ginkgo.BeforeEach(func() {
+			ginkgo.By("Setting up data")
+			configmapToModify := &v1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "my-configmap-to-modify"}, Data: map[string]string{"configmap-key": "configmap-value"}}
+			_, err = f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Create(context.TODO(), configmapToModify, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				framework.ExpectNoError(err, "while creating configmap to modify")
+			}
+		})
+
+		ginkgo.It("should verify that pod can restart successfully after configmaps changed", func() {
+			pod := testsuites.SubpathTestPod(f, "configmap-key", "configmap", &v1.VolumeSource{ConfigMap: &v1.ConfigMapVolumeSource{LocalObjectReference: v1.LocalObjectReference{Name: "my-configmap-to-modify"}}}, privilegedSecurityContext)
+			testsuites.SetInitCommandToTouchProbeFile(pod)
+			configmapModified := &v1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "my-configmap-to-modify"}, Data: map[string]string{"configmap-key": "configmap-modified-value"}}
+			testsuites.TestPodContainerRestart(f, pod, func() error {
+				if _, err := f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Update(context.TODO(), configmapModified, metav1.UpdateOptions{}); err != nil {
+					return err
+				}
+				time.Sleep(10 * time.Second)
```

Comment:
Since kubelet updates the configmap periodically, this has the potential to not be updated by the time the container restarts. Maybe a more reliable method would be to restart the container repeatedly until the modified data is detected?